android/app/src/main/java/safetyfirst/androidapp/safetyfirstcontroller/MainActivity.java

```diff
@@ -145,6 +178,85 @@ public void onCheckedChanged(CompoundButton compoundButton, boolean b) {//Publis
         });
     }
 
+    @SuppressLint("NonConstantResourceId")
+    @Override
+    public boolean onNavigationItemSelected(@NonNull MenuItem item) {
+        switch (item.getItemId()) {
+            case R.id.home:
+                getSupportFragmentManager().beginTransaction().replace(R.id.fragment_container,
+                        new HomeFragment()).commit();
+                break;
+
+            case R.id.profile:
+                getSupportFragmentManager().beginTransaction().replace(R.id.fragment_container,
+                        new ProfileFragment()).commit();
+                break;
+
+            case R.id.contacts:
+                getSupportFragmentManager().beginTransaction().replace(R.id.fragment_container,
+                        new ContactsFragment()).commit();
+                break;
+
+            case R.id.login:
+                getSupportFragmentManager().beginTransaction().replace(R.id.fragment_container,
+                        new LoginFragment()).commit();
+                break;
+
+            case R.id.registerUser:
+                getSupportFragmentManager().beginTransaction().replace(R.id.fragment_container,
+                        new RegisterFragment()).commit();
+                break;
+            case R.id.logOut:
+                FirebaseAuth.getInstance().signOut();
+                hideItemDefault();
+                getSupportFragmentManager().beginTransaction().replace(R.id.fragment_container,
+                        new HomeFragment()).commit();
+                Toast.makeText(MainActivity.this, "Logged Out", Toast.LENGTH_LONG).show();
+        }
+        drawerLayout.closeDrawer(GravityCompat.START);
+        return true;
+    }
+
+    public void hideItemLogged() {
+        NavigationView navigationView = (NavigationView) findViewById(R.id.nav_view);
+        Menu nav_Menu = navigationView.getMenu();
```

Comment:
Naming conventions

Reply:
Here it's referencing an "id" in a XML file, which does not have the same naming conventions as java.

Reply:
nav_Menu in this line is a variable name in Java.

Reply:
Ah yes, my bad can't believe I missed that. What a world we live in.